packages/auth/lib/AuthProviders/AuthProviderOAuth.ts

```diff
@@ -0,0 +1,270 @@
+import { Service, Inject } from "@fullstack-one/di";
+import { SchemaBuilder } from "@fullstack-one/schema-builder";
+import { GraphQl } from "@fullstack-one/graphql";
+import { Auth, AuthProvider } from "..";
+import { Server } from "@fullstack-one/server";
+import { Config } from "@fullstack-one/config";
+import { BootLoader } from "@fullstack-one/boot-loader";
+import * as KoaRouter from "koa-router";
+import * as koaBody from "koa-bodyparser";
+import * as koaSession from "koa-session";
+import * as passport from "koa-passport";
+import { LoggerFactory, ILogger } from "@fullstack-one/logger";
+import { PrivacyAgreementAcceptance } from "../PrivacyAgreementAcceptance";
+
+const template = `
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <script>
+  function go() {
+    var data = <%=data%>;
+
+    for(var i in data.origins) {
+      try {
+        window.opener.postMessage(data.message, data.origins[i]);
+      } catch(e) {}
+    }
+    window.close();
+  }
+
+  window.onload = go;
+  </script>
+  <title>Authetication</title>
+</head>
+<body>
+  <h1>This Page requires Javascript.</h1>
+</body>
+</html>`;
+
+const oAuthCallback = (message, origins) => {
+  const data = {
+    message,
+    origins
+  };
+
+  return template.replace("<%=data%>", JSON.stringify(data));
+};
+
+@Service()
+export class AuthProviderOAuth {
+  private emailAuthProvider: AuthProvider;
+  private oAuthAuthProviders: { [key: string]: AuthProvider } = {};
+
+  private server: Server;
+  private logger: ILogger;
+
+  private authConfig;
```

Comment:
Could we add a `IAuthConfig` interface?

Reply:
This would not make any technical sense. It would just be useful for documentation. However, for me it would help more to create real documentation with more explanation in future than some interface which will never really be used. 

Furthermore, new AuthProviders could need other config which is not in the interface and thus the interface would need something like `IAuthConfig | any`.

Reply:
An interface would help you inside of the `auth` package to know which properties exists.

> Furthermore, new AuthProviders could need other config which is not in the interface and thus the interface would need something like IAuthConfig | any.

Exactly, other AuthProviders may need other interfaces.

Anyhow, I would suggest, that we also use Typescript for our configs and permissions in the long run and there we would need interfaces.

However, since the missing interface is no setback for the `auth` package, I resolve this conversation.